grpclog/loggerv2.go

```diff
@@ -123,16 +123,37 @@ type loggerV2Config struct {
 }
 
 func newLoggerV2WithConfig(infoW, warningW, errorW io.Writer, c loggerV2Config) LoggerV2 {
-	var m []*log.Logger
 	flag := log.LstdFlags
 	if c.jsonFormat {
 		flag = 0
 	}
-	m = append(m, log.New(infoW, "", flag))
-	m = append(m, log.New(io.MultiWriter(infoW, warningW), "", flag))
-	ew := io.MultiWriter(infoW, warningW, errorW) // ew will be used for error and fatal.
-	m = append(m, log.New(ew, "", flag))
-	m = append(m, log.New(ew, "", flag))
+
+	infoLogW := infoW
+
+	// use io.Discard instead of io.MultiWriter if all loggers at a given level are io.Discard.
```

Comment:
Can we just check this condition of all loggers being io.Discard or not. If not, just do new assignments else don't. I think that will make the new code simpler

Reply:
If we just check if all loggers are disabled, then if `Info` is disabled but `Warn` and `Err` aren't, then we won't discard `Info` logs. Or if `Info` and `Warn` are disabled but `Error` isn't, then we won't discard either `Info` or `Warn` logs.

So this would lose all the benefit unless someone has all logging entirely disabled.

Maybe I'm misunderstanding what you're asking?